weblate/trans/machine/microsoft.py

```diff
@@ -178,3 +182,172 @@ def convert_language(self, language):
         if language in self.LANGUAGE_CONVERTER:
             return self.LANGUAGE_CONVERTER[language]
         return language.split('-')[0]
+
+
+class MicrosoftTerminologyService(MachineTranslation):
+    """
+    The Microsoft Terminology Service API.
+
+    Allows you to programmatically access the terminology,
+    definitions and user interface (UI) strings available
+    on the MS Language Portal through a web service (SOAP).
+    """
+    name = 'Microsoft Terminology'
+
+    MS_TM_BASE = 'http://api.terminology.microsoft.com'
+    MS_TM_API_URL = '{base}/Terminology.svc'.format(base=MS_TM_BASE)
+    MS_TM_SOAP_XMLNS = '{base}/terminology'.format(base=MS_TM_BASE)
+    MS_TM_SOAP_HEADER = '{xmlns}/Terminology/'.format(xmlns=MS_TM_SOAP_XMLNS)
+    MS_TM_XPATH = './/{{{xmlns}}}'.format(xmlns=MS_TM_SOAP_XMLNS)
+
+    def soap_req(self, url, http_post=False, skip_auth=False, **kwargs):
+        soap_action = kwargs.get('soap_action', '')
+        url = self.MS_TM_API_URL
+        action = self.MS_TM_SOAP_HEADER + soap_action
+        headers = {
+            'SOAPAction': (
+                '"{action}"'
+            ).format(action=action),
+            'Content-Type': 'text/xml; charset=utf-8'
+        }
+        if soap_action == 'GetLanguages':
+            payload = """
+                <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+                  xmlns="{xmlns}">
+                  <soapenv:Header/>
+                  <soapenv:Body>
+                    <{soap_action}/>
+                  </soapenv:Body>
+                </soapenv:Envelope>
+            """.format(xmlns=self.MS_TM_SOAP_XMLNS, soap_action=soap_action)
+        elif soap_action == 'GetTranslations':
+            source = kwargs.get('source', '')
+            language = kwargs.get('language', '')
+            text = kwargs.get('text', '')
+            max_result = 5
+            if soap_action and source and language and text:
+                payload = """
```

Comment:
#1673  Using `from django.template.loader import get_template `
This next xml could be inside template and use the `get_template('machinery/microsoft_terminology.jinja')` in order to rendener the neede variables like `payload = template.render({'from_lang': 'en-us', 'to_lang': 'es-mx'...})`

Reply:
Certainly using django templates would be better. The current code for example doesn't well handle special chars in the payload.

Reply:
Great will do, @nijel is there a slack channel or gitter I have some issues running the test maybe someone can help me with that, or should I contact you directly through email? Thanks

Reply:
There is Gitter, but honestly it's empty most of the time, that's why it's not advertised anywhere - https://gitter.im/WeblateOrg/Lobby